### PR TITLE
Add LCS implementation

### PR DESCRIPTION
**Pull request by Jay9596**
- Added LCS (Longest Common Subsequence) implementation  
- Added tests for different cases.

### CODE REVIEW

**Comment by AnshulMalik** [comment]
@Jay9596 Love it!, Some comments can really help easily understand the algorithm :)

**Comment by efugier** [comment]
Hey there, nice one.
Please rename it to the full name of the algorithm so that is more self explanatory !

**Comment by Jay9596** [comment]
Made the changes, now the function is called `longest_common_subsequence`.

File: src/dynamic_program/longest_common_subsequence.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    lcs.into_iter().rev().collect()
+}
+
+#[cfg(test)]
+mod test {
+    use dynamic_program::longest_common_subsequence as lcs;
+
+    #[test]
+    fn simple_lcs() {
+        assert_eq!(lcs("ABBADE", "ABCDEF"), ['A', 'B', 'D', 'E']);
```
Comment (AnshulMalik) [inline src/dynamic_program/longest_common_subsequence.rs]: It's be ideal if we get a string back instead of a vector.

**Comment by AnshulMalik** [comment]
Closing due to inactivity.

**Pull request closed without merging**